cardano-api/src/Cardano/Api/Types.hs

```diff
@@ -48,9 +49,19 @@ data KeyPair
   deriving (Generic, NFData, Show)
   deriving anyclass NoUnexpectedThunks
 
-newtype ShelleyKeyDiscriminator = ShelleyKeyDiscriminator KeyDiscriminator
-  deriving (Generic, Show)
-  deriving newtype (NFData, NoUnexpectedThunks)
+-- | A means of discriminating between different kinds of Shelley keys.
+data ShelleyKeyDiscriminator
+  = GenesisShelleyKey
+  | RegularShelleyKey
+  deriving (Generic, NFData, Show)
+  deriving anyclass NoUnexpectedThunks
+
+-- | Convert a 'ShelleyKeyDiscriminator' to a 'KeyDiscriminator'.
+toShelleyLedgerKeyDiscriminator :: ShelleyKeyDiscriminator -> KeyDiscriminator
+toShelleyLedgerKeyDiscriminator skd =
+  case skd of
+    GenesisShelleyKey -> Genesis
+    RegularShelleyKey -> Regular
```

Comment:
Do we want conversion functions in the `Types` module or should they be placed elsewhere?

Reply:
Its fine where it is.

Reply:
This is unused, my bad. Dropping it from the changeset.